packages/apps/graph-client/src/context/chain-tree-context.tsx

```diff
@@ -1,82 +0,0 @@
-import type { IBlock } from '@utils/hooks/use-parsed-blocks';
-import type { ReactElement } from 'react';
-import React, { createContext, useContext, useState } from 'react';
-
-interface IChainTree {
-  [chain: string]: Record<string, IBlock>;
-}
-
-interface IChainTreeContext {
-  chainTree: IChainTree;
-  addBlockToChain: (block: IBlock) => void;
-}
-
-const ChainTreeContext = createContext<IChainTreeContext | undefined>(
-  undefined,
-);
-
-export function addBlockToChainTree(
-  block: IBlock,
-  chainTree: IChainTree,
-): IChainTree {
-  let currentBlock = block;
-
-  //Check if chain index exists, if not create it
-  if (chainTree[currentBlock.chainId] === undefined) {
-    chainTree[currentBlock.chainId] = {};
-  }
-
-  //Add block to chain
-  chainTree[currentBlock.chainId][currentBlock.hash] = currentBlock;
-
-  //Check if parent block exists, if it does increment confirmation depth
-  while (
-    currentBlock.parentHash &&
-    chainTree[currentBlock.chainId][currentBlock.parentHash]
-  ) {
-    const childsConfirmationDepth = currentBlock.confirmationDepth;
-
-    //set current block to parent block
-    currentBlock = {
-      ...chainTree[currentBlock.chainId][currentBlock.parentHash],
-    };
-
-    //before incrementing depth check if it is already 6
-    if (currentBlock.confirmationDepth < 6) {
-      currentBlock.confirmationDepth = childsConfirmationDepth + 1;
-    } else {
-      break;
-    }
-
-    //set the updated block in the chain
-    chainTree[currentBlock.chainId][currentBlock.hash] = currentBlock;
-  }
-
-  return chainTree;
-}
```

Comment:
This is the function that I mentioned. We could create an in-file IBlock definition, so we don't have dependencies